### ISSUE TITLE
Default x-operation-name To Controller Method Name

### ISSUE THREAD

**Issue by CatGuardian** [nestjs/swagger#584]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Currently the generated operationId in the OpenAPI spec follows this pattern: `<Controller Class Name>_<Controller Class Method Name>`.

This in itself is probably a good thing. But then when I use [ng-openapi-gen](https://github.com/cyclosproject/ng-openapi-gen) it will generate a method from the exact operationId.

So for my `findAll()` GET method of my  `CatsController` class it will generate me an Angular service with the following method name: `catsControllerFindAll()`.

## Expected behavior

The current behavior is undesirable because the `catsControllerFindAll()` method belongs to my `CatsApiService`. So it would be much better if `findAll()` method would be generated in my `CatsApiService`

Fortunately `ng-openapi-gen` can support this using the `x-operation-name` tag. [See here about the x-operation-name tag](https://github.com/cyclosproject/ng-openapi-gen#supported-vendor-extensions)

So what I am asking is the following:
* By default, set the `x-operation-name` tag to be just the name of the method. So for my `findAll()` GET method of my `CatsController`, the x-operation-name tag would have a value of `findAll`.
* Add `xOperationName` to the options object of the `@ApiOperation()` decorator. And if the `xOperationName` is specified, then set the `x-operation-name` tag to the value specified.

## What is the motivation / use case for changing the behavior?

I kind of already explained the motivation with my previous example. The motivation is to make the generated API services in Angular spit out more friendly method names.

## Environment

<pre><code>
Nest version: 6.14.2

**Comment by CatGuardian** [comment]
@kamilmysliwiec If I wanted to work on a pull request for this, then is it safe to assume that we would want to set the x-operation-name tag by default to the method name? Or does that go against some development principal and we only want to add it to the ApiOperation decorator?

My opinion is that what harm would be done by defaulting it to the method name? As long as it can be changed by the ApiOperation Decorator.

**Comment by kamilmysliwiec** [comment]
We can make this configurable. Would you like to create a PR for this?

**Comment by KerryRitter** [comment]
@kamilmysliwiec I believe @nartc's update resolves this and this can be closed:

![image](https://user-images.githubusercontent.com/2314053/98551862-6ebe3d00-2263-11eb-9ab3-47d46ad09722.png)

**Comment by kamilmysliwiec** [comment]
Great catch @KerryRitter, thanks!

**Comment by Vegita2** [comment]
As far as I understand this only allows you to override the `operationId` not actually set `x-operation-name`. Per specification `operationId` must be unique among all operations described in the API. That's why `ng-openapi-gen` uses `x-operation-name` to override the method names instead so they can have duplicate names.

Is this currently possible?

**Comment by Vegita2** [comment]
For anyone else having the same problems, there is actually a way to specify the `x-operation-name` property. NestJS provides the decorator `@ApiExtension()` which adds the specified property to the method. So to automatically add it for every method in a class with the value as the method name it needs to be wrapped in a class decorator:

```typescript
import { ApiExtension } from '@nestjs/swagger';

export function AutoOperationName(): ClassDecorator {
  return (target: any) => {
    for (const key of Object.getOwnPropertyNames(target.prototype)) {
      let descriptor = Object.getOwnPropertyDescriptor(target.prototype, key);
      if (descriptor) {
        const extension = ApiExtension('x-operation-name', key);
        descriptor = extension(target, key, descriptor) as TypedPropertyDescriptor<any>;
        Object.defineProperty(target.prototype, key, descriptor);
      }
    }
  };
}
```

Which can be used like this and it will add `x-operation-name` to every method with the method name
```typescript
@Controller('cats')
@AutoOperationName()
export class CatsController {
...
}
```

**Comment by aleix10kst** [comment]
I've seen this has been fixed here: [#747 ](https://github.com/nestjs/swagger/pull/747)
